workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -40,24 +41,29 @@ public SpatialOSBehaviourLibrary(ILogDispatcher logger)
             this.logger = logger;
         }
 
-        public void InjectAllReadersWriters(MonoBehaviour spatialOSBehaviour)
+        public Dictionary<uint, IReaderInternal> InjectAllReadersWriters(MonoBehaviour spatialOSBehaviour, Entity entity)
         {
             var spatialOSBehaviourType = spatialOSBehaviour.GetType();
             EnsureLoaded(spatialOSBehaviourType);
+            var createdReaderWriters = new Dictionary<uint, IReaderInternal>();
             if (invalidMonoBehaviourTypes.Contains(spatialOSBehaviourType))
             {
-                return;
+                return createdReaderWriters;
             }
 
             foreach (var readerWriterComponentId in componentReaderIdsForBehaviours[spatialOSBehaviourType])
             {
-                Inject(spatialOSBehaviour, readerWriterComponentId);
+                var readerWriter = Inject(spatialOSBehaviour, readerWriterComponentId, entity);
+                createdReaderWriters[readerWriterComponentId] = readerWriter;
             }
 
             foreach (var readerWriterComponentId in componentWriterIdsForBehaviours[spatialOSBehaviourType])
```

Comment:
What happens here if you require both a reader and a writer of the same component?

Reply:
That is invalid because you are meant to use Writers in place of Readers as well, so you get an error when you try to inject (see from line 155) and the MonoBehaviour will never be enabled. (This matches the SDK behavior btw.)

Reply:
it's been a while since the last design meeting, so I might be wrong, but did we not decide to allow this behaviour?

Reply:
Yeah, we backtracked on the design meeting decision here. The design meeting decision was indeed to allow for multiple readers/writers per component per monobehaviour. Sorry for the backtracking here. I didn't communicate that properly. I can't remember the specifics anymore but something related to not being able to clearly identify a reader/writer field in a behaviour when given a just component id made the original semantics really hard to implement. Now looking at the code again, I can't see/remember anymore what the problem was so I'm happy with us reverting back to the initial design meeting decision. At the same time, this is not a matter that I feel very strongly about so I'd be fine with either outcome (no restrictions vs. at most one reader/writer per monobehaviour).